cmd/if-mover/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+const (
+	ifBase     = "mlink"
+	hostNsFile = "/tmp/hostns.txt"
+	multusLink = "net1"
+)
+
+func main() {
+
+	log.Println("Determining host namespace id")
+	// The Host network namespace name comes from a file generated by the following command:
+	// ip --json netns list-id | jq -j ".[] | select(.nsid == 0).name" > /tmp/hostns.txt
+	// It's expected that this was run before this program runs.
+	// This was the most consistent way I found to convey the host namespace to a golang program.
+
+	hostNsBytes, err := ioutil.ReadFile(hostNsFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	hostNsName := string(hostNsBytes)
+
+	// Original namespace is kept for switching back
+	log.Println("Getting namespace handles")
+
+	origNs, err := netns.Get()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer origNs.Close()
+
+	hostNs, err := netns.GetFromName(hostNsName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer hostNs.Close()
+
+	log.Println("Switching to host namespace to determine the new interface name")
+	// Note from github.com/vishvananda/netlink documentation:
+	// "
+	// After locking a goroutine to its current OS thread with
+	// runtime.LockOSThread() and changing its network namespace,
+	// any new subsequent goroutine won't be scheduled on that
+	// thread while it's locked. Therefore, the new goroutine will
+	// run in a different namespace leading to unexpected results.
+	// "
+	// This won't be a problem here as long as the program stays as 1 goroutine.
```

Comment:
Where is the goroutine?

Reply:
The main goroutine running the main function.